src/NHibernate/Engine/BatchFetchQueue.cs

```diff
@@ -506,10 +505,10 @@ bool ProcessKey(EntityKey key, bool ignoreCache = false)
 						return false;
 					}
 
-					if (!checkCache || !IsCached(key, persister))
-					{
-						ids[i++] = key.Identifier;
-					}
+					// No need to check "!checkCache || !IsCached(key, persister)": "batchableCache == null"
+					// already means there is no cache, so IsCached can only yield false. (This method is now
+					// removed.)
+					ids[i++] = key.Identifier;
 				}
```

Comment:
The same correction should also be done in `GetCollectionBatch` method as it has the same logic.

Reply:
Fixed.